7a_temp_coop_munge/src/data_parsers/parse_missouri_files.R

```diff
@@ -0,0 +1,85 @@
+
+# Parse the ~300 files within the `Bull_Shoals_Lake_DO_and_Temp.zip` file
+
+parse_Bull_Shoals_Lake_DO_and_Temp <- function(inind, outind) {
+
+  infile <- sc_retrieve(inind, remake_file = '6_temp_coop_fetch_tasks.yml')
+  outfile <- as_data_file(outind)
+
+  # unzip the file with all of Bull Shoals
+  files_from_zip <- unzip(infile, exdir = tempdir())
+
+  # Read and clean each file (create on big file)
+  data_clean <- purrr::map(files_from_zip, function(fn) {
+
+    # Only read in the data part of the Excel file
+    # Assumes each file has only these rows/columns
+    # of data (which looked correct based on my spot check)
+    readxl::read_excel(fn, range = "A7:D27") %>%
+
+      # Struggles with the special symbols in the column names
+      # due to encoding issues. See details in the `scipiper`
+      # issue: https://github.com/USGS-R/scipiper/issues/151
+      fix_colname_encoding("latin1") %>%
+
+      # Reformat columns and extract info we need from Excel file
+      dplyr::mutate(DateTime = extract_bullshoals_date(fn),
+                    site = extract_bullshoals_site(fn),
+                    # TODO: Don't manually add the nhdhr id
+                    # Only doing now bc we don't have hypso for
+                    # Bull Shoals yet and I found this nhdhr match:
+                    # https://github.com/USGS-R/lake-temperature-model-prep/blob/master/4_params_munge.yml#L60
+                    id = 'nhdhr_12003253',
+                    depth = `Depth (ft)`) %>%
+      dplyr::rename(temp = "Temp (°C)") %>%
+
+      # Replace "Surface" depth and convert depths to numeric
+      mutate(depth = as.numeric(ifelse(depth == "Surface", 0, depth))) %>%
+
+      # Change depth from feet to meters
+      mutate(depth = depth / 3.28) %>%
```

Comment:
elsewhere we multiply feet by 0.3048, which is a bit different than dividing by 3.28. 3.28 is the rounded version of this, and should be more like 3.28084. 

I think it is fine to hard-code it here, but in the future we may want some kind of general "utils.R" set of functions for things like unit conversions that can happen throughout the pipeline and may need to be functions(?). This is such a simple one I don't think it is necessary, but since this conversion is different than how it is done elsewhere, it caught my eye.

Reply:
Are you referring to the `unit_map`s I am seeing elsewhere in the repo? Such as [here](https://github.com/USGS-R/lake-temperature-model-prep/blob/5b934a95f83b9bdcc50fa4704a8817f58de8328b/7a_wqp_munge/src/munge_wqp_dat.R#L12-L14). I see those hardcoded in multiple places, so agree that standardizing would be good. I think Julie is having to convert temperature units and so having helper functions to do that may make sense sooner rather than later.

Reply:
Yes, that is one of the places where it shows up.

Reply:
I created #235

Reply:
I will pick this one off when I add the Norfolk dataset (which will require this conversion)